04_numeronym/marchitect/main.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	result := make([]string, 0)
+	for _, val := range vals {
+		numeronym := ""
+		length := len([]rune(val))
```

Comment:
Hold on to []rune conversion as you use it again down below, e.g.:
```go
runes := []rune(val)
```
(conversion is O(N), N being the number of chars).